src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/client/BitbucketServerClient.java

```diff
@@ -122,49 +108,15 @@ public boolean supportsCodeInsights() {
         return false;
     }
 
-    private void validate(Response response) throws IOException {
-        if (!response.isSuccessful()) {
-            ErrorResponse errors = null;
-            if (response.body() != null) {
-                errors = getObjectMapper().reader().forType(ErrorResponse.class)
-                        .readValue(response.body().string());
-            }
-            throw new BitbucketException(response.code(), errors);
-        }
-    }
-
     private OkHttpClient getClient() {
-        client = Optional.ofNullable(client).orElseGet(() ->
-                new OkHttpClient.Builder()
-                        .authenticator(((route, response) ->
-                                response.request()
-                                        .newBuilder()
-                                        .header("Authorization", format("Bearer %s", getToken()))
-                                        .header("Accept", APPLICATION_JSON_MEDIA_TYPE.toString())
-                                        .build()
-                        ))
-                        .build()
-        );
-        return client;
-    }
-
-    private ObjectMapper getObjectMapper() {
-        objectMapper = Optional.ofNullable(objectMapper).orElseGet(() -> new ObjectMapper()
-                .setSerializationInclusion(Include.NON_NULL)
-                .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
-        );
-        return objectMapper;
-    }
-
-    private String baseUrl() {
-        return configuration.get(BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_URL)
-                .orElseThrow(() ->
-                        new IllegalArgumentException(format("Missing required property %s", BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_URL))
-                );
-    }
-
-    private String getToken() {
-        return configuration.get(BitbucketServerPullRequestDecorator.PULL_REQUEST_BITBUCKET_TOKEN)
-                .orElseThrow(() -> new IllegalArgumentException("Personal Access Token for Bitbucket Server is missing"));
+        return new OkHttpClient.Builder()
+                .addInterceptor(chain -> {
```

Comment:
@mc1arke The code before used the authenticator call. However, the header for the authorization was never passed to the actual requests thus I changed it to interceptor based.

Reply:
👍